sys/parsoid_proxy.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+/*
+ * Simple proxy to route requests to the client-requested
+ * Parsoid variant (JS or PHP) during the transition
+ * period. Parsoid/JS is being phased out and replaced by
+ * Parsoid/PHP.
+ */
+
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+const mwUtil = require('../lib/mwUtil');
+const VARIANT_HDR_NAME = 'x-parsoid-variant';
+
+module.exports = () => {
+    return {
+        spec: {
+            paths: {
+                '/{+path}': {
+                    all: {
+                        operationId: 'proxy_parsoid_variant'
+                    }
+                }
+            }
+        },
+        operations: {
+            proxy_parsoid_variant: (hyper, req) => {
+                const rootReqHeaders = hyper._rootReq.headers || {};
+                if (!rootReqHeaders[VARIANT_HDR_NAME]) {
+                    rootReqHeaders[VARIANT_HDR_NAME] = 'JS';
+                }
```

Comment:
Using `||` is shorter and nicer then a full if.

Reply:
Agreed. Changed it.